server/src/cmds/config/dirs.rs

```diff
@@ -216,19 +218,47 @@ pub fn top_dialog(db: &Arc<db::Database>, siv: &mut Cursive) {
     );
 }
 
+fn tab_completer(content: &str) -> Box<[String]> {
```

Comment:
tiny style nit: I default to `Vec<...>`. The `collect` is probably more efficient on that than `Box<[...]>`, at least when the iterator doesn't know the size in advance; converting to box means a final copy to shrink the capacity down to the size. I only tend to use the boxed slice form when I'm constructing them infrequently and might be keeping millions of them around (so I want to avoid keeping the extra `usize`).

Reply:
I checked the `FromIterator` implementations on `Vec<_>` and `Box<[_]>` in the standard library to verify this, and FromIterator for `Box<[_]>` just collects into a `Vec` and then calls `.into_boxed_slice()` (which would have to reallocate if it didn't allocate exactly the right amount). Fair point on vectors being more efficient here.